client/src/components/Dataset/DatasetDisplay.vue

```diff
@@ -0,0 +1,101 @@
+<script setup lang="ts">
+import { BAlert } from "bootstrap-vue";
+import { storeToRefs } from "pinia";
+import { computed, ref, watch } from "vue";
+
+import { useDatasetStore } from "@/stores/datasetStore";
+import { useUserStore } from "@/stores/userStore";
+import { withPrefix } from "@/utils/redirect";
+import { errorMessageAsString } from "@/utils/simple-error";
+import { bytesToString } from "@/utils/utils";
+
+import Alert from "@/components/Alert.vue";
+import LoadingSpan from "@/components/LoadingSpan.vue";
+import CenterFrame from "@/entry/analysis/modules/CenterFrame.vue";
+import TabularChunkedView from "components/Dataset/Tabular/TabularChunkedView.vue";
+
+interface Props {
+    datasetId: string;
+    isBinary: boolean;
+}
+
+const { getDataset, isLoadingDataset } = useDatasetStore();
+
+const props = defineProps<Props>();
+
+const contentTruncated = ref();
+const contentChunked = ref();
+const errorMessage = ref();
+const sanitizedJobImported = ref();
+const sanitizedToolId = ref();
```

Comment:
Please consider using proper types for better type safety.

```suggestion
const contentTruncated = ref<number | null>(null);
const contentChunked = ref<boolean>(false);
const errorMessage = ref<string>("");
const sanitizedJobImported = ref<boolean>(false);
const sanitizedToolId = ref<string | null>(null);
```